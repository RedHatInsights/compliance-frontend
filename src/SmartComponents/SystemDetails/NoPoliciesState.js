import React from 'react';
import { BackgroundLink } from 'PresentationalComponents';
import { PlusCircleIcon } from '@patternfly/react-icons';
import {
  Title,
  Bullseye,
  Button,
  EmptyState,
  EmptyStateBody,
  EmptyStatePrimary,
  EmptyStateSecondaryActions,
  EmptyStateIcon,
} from '@patternfly/react-core';

const NoPoliciesState = () => (
  <Bullseye>
    <EmptyState>
      <EmptyStateIcon icon={PlusCircleIcon} />
      <Title headingLevel="h1" size="lg">
        This system is not part of any SCAP policies defined within Compliance.
      </Title>
      <EmptyStateBody>
        To assess and monitor compliance against a SCAP policy for this system,
        add it to an existing policy or create a new policy.
      </EmptyStateBody>
      <EmptyStatePrimary>
        <BackgroundLink to="/scappolicies/new">
          <Button variant="primary" ouiaId="CreateNewPolicyButton">
            Create new policy
          </Button>
        </BackgroundLink>
      </EmptyStatePrimary>
      <EmptyStateSecondaryActions>
        <Button
          variant="link"
          component="a"
          href="/insights/compliance/scappolicies"
        >
          View compliance policies
        </Button>
      </EmptyStateSecondaryActions>
    </EmptyState>
  </Bullseye>
);

export default NoPoliciesState;
