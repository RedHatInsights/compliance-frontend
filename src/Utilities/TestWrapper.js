import propTypes from 'prop-types';
import { MemoryRouter } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import TableStateProvider from '@/Frameworks/AsyncTableTools/components/TableStateProvider';

import { Provider } from 'react-redux';
import { init } from 'Store';

const TestWrapper = ({ children, mocks = [], routes, store: propStore }) => {
  const store = init().getStore();

  return (
    <MemoryRouter {...(routes ? { initialEntries: routes } : {})}>
      <TableStateProvider>
        <Provider store={propStore || store}>
          <MockedProvider mocks={mocks} addTypename>
            {children}
          </MockedProvider>
        </Provider>
      </TableStateProvider>
    </MemoryRouter>
  );
};

TestWrapper.propTypes = {
  children: propTypes.node,
  mocks: propTypes.array,
  routes: propTypes.array,
  store: propTypes.object,
};

export default TestWrapper;
