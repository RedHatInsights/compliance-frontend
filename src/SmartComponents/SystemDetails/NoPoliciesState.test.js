import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import TestWrapper from '@/Utilities/TestWrapper';

import NoPoliciesState from './NoPoliciesState';

describe('NoPoliciesState', () => {
  it('renders a note with a system having no policies', () => {
    render(
      <TestWrapper>
        <NoPoliciesState />
      </TestWrapper>
    );

    expect(
      screen.getByText(
        'This system is not part of any SCAP policies defined within Compliance.'
      )
    ).toBeInTheDocument();
  });
});
