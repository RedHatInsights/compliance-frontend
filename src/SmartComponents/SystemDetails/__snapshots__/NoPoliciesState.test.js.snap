// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoPoliciesState with a system having policies 1`] = `
<Bullseye>
  <EmptyState>
    <EmptyStateIcon
      icon={[Function]}
    />
    <Title
      headingLevel="h1"
      size="lg"
    >
      This system is not part of any SCAP policies defined within Compliance.
    </Title>
    <EmptyStateBody>
      To assess and monitor compliance against a SCAP policy for this system, add it to an existing policy or create a new policy.
    </EmptyStateBody>
    <EmptyStatePrimary>
      <BackgroundLink
        to="/scappolicies/new"
      >
        <Button
          ouiaId="CreateNewPolicyButton"
          variant="primary"
        >
          Create new policy
        </Button>
      </BackgroundLink>
    </EmptyStatePrimary>
    <EmptyStateSecondaryActions>
      <Button
        component="a"
        href="/insights/compliance/scappolicies"
        variant="link"
      >
        View compliance policies
      </Button>
    </EmptyStateSecondaryActions>
  </EmptyState>
</Bullseye>
`;
